### DISCUSSION TITLE
마이페이지와 사용자의 로그인 상태 유지에 대해

### DISCUSSION THREAD

**Discussion by hozzijeong** [woowacourse-teams/2023-pium#253] [뜨거운 토론]
### 안건
`me` API를 도입해야하나?

금일(8월 11일) 아침에 `me` API를 도입하지 않고 마이페이지 클릭 시에 로그인 하지 않은 상태라면 로그인 페에지로, 그게 아니라면 마이페이지로 이동하기로 했습니다. 하지만 곰곰이 생각을 해보니 '마이페이지에서 **로그인 하지 않은 상태** 는 어떻게 구분할까?' 하는 의문이 생겼습니다. 

마이 페이지에 들어갔을 때 그 어떠한 API도 호출하지 않기 때문에 현재 세션이 유효한지 아닌지 알 수 없습니다. 따라서, 마이페이지에 들어왔을 때 사용자 정보를 요청하는 API를 만든다면, 해당 문제는 잠시나마 해결이 가능합니다.

하지만 또다시 비슷한 문제를 마주 했을 때 다른 API를 요청하는 방식으로 문제를 해결할 수 있을것 같지는 않습니다. 따라서 현재 사용자의 세션 유효성을 확인하는 `me` API가 필요하다고 생각하는데, 다른 팀원들의 생각은 어떤지 궁금합니다.

만약에 `me` API가 생긴다면 마이페이지에 들어왔을 때 사용자 정보 요청 API를 따로 만들지 않아도 됩니다. 아마 추가적으로 client에서도 사용자의 로그인 여부를 나타내는 변수를 만들어서 관리할 것 같습니다.

**Comment by Choi-JJunho** [comment]
마이페이지 클릭 시에 로그인 하지 않은 상태에서 사용자에게 제공하는 정보가 없다면 골치아프겠군요...
향후 확장성을 생각해 봤을 때나 현재 사용자에게 마이페이지로 제공할만한 정보가 없다는 부분을 생각해 봤을 때 `/members/me` API를 추가하는 방향이 좋을 것 같습니다.

추가적인 의견을 덧붙이자면 모든 API에서 401 에러가 발생하는 것을 핸들링 하는 것도 좋지만 `/members/me` API 로 공통된 검증을 수행한다면 코드레벨에서도 좋은 개선방향을 기대할 수 있겠네요 (프론트 코드는 잘 모르지만..)

**Reply by hozzijeong** [reply]
공통 검증을 한다는 거라면, `/members/me`를 호출하고 `/reminder`를 호출하는 방식일까요?

**Reply by Choi-JJunho** [reply]
넵 말씀하신 방향이 제가 말한 방식이긴하나 어디까지나 추측이라.. 
/members/me를 도입하는 시점에서부터는 프론트분들께서 상태관리하기 편한 방향으로 진행하시는것이 좋을 것 같아용!

**Comment by WaiNaat** [comment]
1. 리마인더나 내 반려식물 목록 같은 경우에는 세션이 만료되었다면 접속해서 자료를 받아올 때 터지겠지만, 마이페이지는 사실 서버한테 받아오는 정보가 아예 없어서 로그인 여부를 알기 힘들어요. 새로고침 시 딱 한 번 요청을 보내서 로그인 상태인지 확인하는 작업이 있어야 할 것 같아요
2. 저희가 맨 처음에 기획할 때 "로그인한 사용자의 첫 화면은 리마인더이다"라는 내용을 이야기했었는데요. 이걸 적용한다면 사용자가 로그인한 상태인지 확인하는 api가 필요하다고 생각해요!

**Reply by WaiNaat** [reply]
가능할지는 모르겠지만 해당 요청을 보냈을 때 세션 유지시간을 다시 6시간으로 설정하면 더 좋을 것 같아요
추가적으로 이런 api를 만들어서 사용한다면 내비게이션 바를 유동적으로 로그인/마이페이지로 바꾸는 것도 괜찮아 보입니다

**Reply by yeonkkk** [reply]
특정 요청을 보냈을 때 세션을 다시 생성해서 연장(유지)하는 방법 가능합니다.
해당 api에 대한 결론이 나면 고려해서 구현해도 좋을 것 같네요.

**Comment by Kim0914** [comment]
저도 위 두분 의견과 동일합니다 !

`/members/me`를 추가하는 방법이 좋아보여요.

**Comment by yeonkkk** [comment]
저도 해당 api 추가하는 것 좋습니다!
